rewrite-gradle/src/main/java/org/openrewrite/gradle/RemoveEnableFeaturePreview.java

```diff
@@ -0,0 +1,69 @@
+package org.openrewrite.gradle;
+
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.groovy.GroovyIsoVisitor;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.J.MethodInvocation;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class RemoveEnableFeaturePreview extends Recipe {
+
+  private static final String ENABLE_FEATURE_PREVIEW_METHOD_NAME = "enableFeaturePreview";
+
+  @Option(displayName = "The feature preview name",
+      description = "The name of the feature preview to remove.",
+      example = "ONE_LOCKFILE_PER_PROJECT"
+  )
+  public String previewFeatureName;
+
+  @Override
+  public @NotNull String getDisplayName() {
+    return "Remove a Gradle enableFeaturePreview method";
+  }
+
+  @Override
+  public @NotNull String getDescription() {
+    return "Remove a Gradle enableFeaturePreview method from settings.gradle / settings.gradle.kts.";
+  }
+
+  @Override
+  public TreeVisitor<?, ExecutionContext> getVisitor() {
+    return Preconditions.check(
+        new IsSettingsGradle<>(),
+        new RemoveEnableFeaturePreviewVisitor());
+  }
+
+  public class RemoveEnableFeaturePreviewVisitor extends GroovyIsoVisitor<ExecutionContext> {
+
+    @Override
+    public MethodInvocation visitMethodInvocation(MethodInvocation method,
+        @NotNull ExecutionContext executionContext) {
+      if (ENABLE_FEATURE_PREVIEW_METHOD_NAME.equals(method.getSimpleName())) {
```

Comment:
If I recall correctly, the type information should be present such that this can be done with a `MethodMatcher`.

```
MethodMatcher enableFeaturePreviewMatcher = new MethodMatcher("RewriteSettings enableFeaturePreview(String)");
if (enableFeaturePreviewMatcher.matches(m)) {
    // do the thing
}
```
https://github.com/openrewrite/rewrite/blob/main/rewrite-gradle/src/main/groovy/RewriteSettings.groovy#L32-L36

Reply:
The pattern above didn't work, but this one below did:
```
new MethodMatcher("org.gradle.api.initialization.Settings enableFeaturePreview(String)");
```
I debugged and constructed a `MethodMatcher` by passing in the actual method object, then looked at the toString() and saw this. Let me know if I should do something different.

Reply:
That works, you could have also flipped the inheritance included parameter to `true` as well. I forgot that in my example, but this does the same thing.